Automattic/jetpack
# Protect: Add WAF standalone mode indicator

### Description
Adds an indicator within the `FirewallHeader` when standalone mode is enabled.

Related Jetpack PR: https://github.com/Automattic/jetpack/pull/34854

**Note**: I've used a common `Status` component with a `label={ __( Standalone mode, slug ) }` as the indicator across both plugins but I am open to suggestions as to other, more attractive or functional ways of doing so.

## Proposed changes:

* Add an initial state variable that retrieves the value of a constant set when standalone mode is activated
* Add a condition to the `FirewallHeader` to indicate when the mode is active
* Replace our custom status `Text` element with the js-packages `Status` component
* Fix the `currentStatus` logic so the FirewallHeader content is updated appropriately

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/204

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch
* Start your Jurassic Tube
* Enable Protect and verify that the WAF is `Active` from the Firewall tab
* Update the `auto_prepend_file` PHP directive to point to `wp-content/jetpack-waf/bootstrap.php` to activate standalone mode
* Return to the Firewall tab and confirm that the active status is replaced with `Standalone mode`
* Enable and disable various WAF settings, deactivate the module and view the Firewall section from an unsupported env to ensure that the conditional FirewallHeader content is updated as expected

## Screenshots
WAF active, Standalone mode disabled:
<img width="596" alt="Screen Shot 2024-01-04 at 11 35 24" src="https://github.com/Automattic/jetpack/assets/43220201/a6d37344-6c80-4e74-8dc5-03ce68652727">

WAF active, Standalone mode enabled:
<img width="605" alt="Screen Shot 2024-01-04 at 11 34 24" src="https://github.com/Automattic/jetpack/assets/43220201/12ff6152-7047-4213-9223-650507b0f76a">

WAF inactive:
<img width="582" alt="Screen Shot 2024-01-04 at 11 36 48" src="https://github.com/Automattic/jetpack/assets/43220201/0656660e-2abc-40cd-9c80-4f7863a70466">
